llm-models/embedding/automl/automl_embeddings_example.py

```diff
@@ -0,0 +1,147 @@
+# Databricks notebook source
+# MAGIC %md # AutoML Embeddings with LangChain
+# MAGIC
+# MAGIC LangChain is a tool that allows you to easily connect large language models with vector stores. In this specific example, we are assuming we have trained an embeddings model with AutoML Embeddings and have the `run_id` of the experiment on hand.
+# MAGIC
+# MAGIC Requirements
+# MAGIC - Databricks Runtime ML 13.1 and above
+# MAGIC - (Recommended) GPU instances
+
+# COMMAND ----------
+
+# MAGIC %pip install -q faiss-cpu
+# MAGIC dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# MAGIC %md #### Note on Dataset Preparation and Inference
+# MAGIC
+# MAGIC It is highly recommended that you preprocess the dataset that you want to use for inference. Here are some recommendations:
+# MAGIC
+# MAGIC 1. Even before fine-tuning the model, it is highly recommended that you remove any non-plain-text elements, like HTML tags or reStructuredText elements. We want to ideally have clean, easily readable text, for AutoML Embeddings.
+# MAGIC 2. Chunk the text contents of the table to ~500 tokens or ~1500 characters. For example, using the following code:
+# MAGIC
+# MAGIC ```
+# MAGIC df = spark.read.table(<table name>)
+# MAGIC
+# MAGIC # User-defined function to split a string into chunks of a given length
+# MAGIC def chunk_string(s, chunk_size=1500):
+# MAGIC     return [s[i:i+chunk_size] for i in range(0, len(s), chunk_size)]
+# MAGIC
+# MAGIC chunk_string_udf = udf(chunk_string, ArrayType(StringType()))
+# MAGIC
+# MAGIC # Split each string and explode the resulting arrays to create new rows
+# MAGIC df_chunked = df.withColumn("text_chunks", chunk_string_udf(col("text"))) \
+# MAGIC                .withColumn("chunk", explode(col("text_chunks"))) \
+# MAGIC                .drop("text", "text_chunks")
+# MAGIC
+# MAGIC df_chunked.write.saveAsTable(<updated table name>)
+# MAGIC ```
+# MAGIC
+# MAGIC 3. Prefix each chunk with the text "passage: ", which is highly recommended for embedding models from the [e5-suite](https://huggingface.co/intfloat/e5-base-v2). For example, using the following code:
+# MAGIC ```
+# MAGIC df = spark.read.table(<table name>)
+# MAGIC
+# MAGIC df_prefixed = df.withColumn("text", concat(lit("passage: "), col("text")))
+# MAGIC
+# MAGIC df_prefixed.write.saveAsTable(<updated table name>)
+# MAGIC ```
+# MAGIC 4. It is recommended to prefix each of your queries with the text "query: " for optimal performance.
+
+# COMMAND ----------
+
+# The run_id from the AutoML Embeddings experiment
+run_id = ""
```

Comment:
I would add some TODO text in the strings to make it absolutely clear that the user needs to do this.
Or, add a comment saying "Fill out these variables according to your automl experiment" or something like that.